SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -436,7 +449,63 @@ theorem Lets.denote_getIExpr {Γ₁ Γ₂ : Ctxt Ty} : {lets : Lets Op Γ₁ Γ
   . contradiction
   . rw[←Option.some_inj.mp he, denote_getIExprAux]
 
+/-
+  ## Mapping
+  We can map between different dialects
+-/
+
+section Map
 
+instance : Functor Signature where
+  map := fun f ⟨sig, regSig, outTy⟩ => 
+    ⟨f <$> sig, regSig.map (fun ⟨a, b⟩ => ⟨f <$> a, f b⟩), f outTy⟩
+
+/-- A dialect morphism consists of a map between operations and a map between types, 
+  such that the signature of operations is respected
+-/
+structure DialectMorphism (Op Op' : Type) {Ty Ty' : Type} [OpSignature Op Ty] [OpSignature Op' Ty'] where
```

Comment:
@ChrisHughes24 what's the `mathlib` way of defining morphisms? I know that they have some notion of morphism?

Reply:
Note: this should not hold the PR from being merged; I want to refactor it post-merge perhaps to use mathlib Morphism tuff.

Reply:
Note that before defining `ICom` or `IExpr` as a category, we have to define the morphisms in the first place, so I doubt that mathlib will help us state this more succinctly

Reply:
No doubt :) I was hoping to get more notation / tooling *once* we define the morphism.